uportal-war/src/main/java/org/jasig/portal/api/permissions/ApiPermissionsService.java

```diff
@@ -79,10 +79,10 @@
             	}
             }
         }
-        
+
         if (includeInherited) {
             IGroupMember member = GroupService.getGroupMember(authP.getKey(), authP.getType());
-            for (@SuppressWarnings("unchecked") Iterator<IEntityGroup> iter = member.getAncestorGroups(); iter.hasNext();) {
+            for (Iterator<IEntityGroup> iter = member.getAncestorGroups().iterator(); iter.hasNext();) {
                 IEntityGroup parent = iter.next();
 
```

Comment:
I totally agree.  As far as the non-GaP code that was affected by this API change, I was moving pretty quickly.  This effort is volunteerism.  I'm passionate that this kind of reboot is needed.